taskvine/src/manager/vine_manager.c

```diff
@@ -4289,6 +4350,7 @@ static struct vine_task *vine_wait_internal(struct vine_manager *q, int timeout,
 		// return if manager is empty and something interesting already happened
 		// in this wait.
 		if(events > 0) {
+			if(task_state_any(q, VINE_TASK_RETRIEVED) && t == NULL) continue;
```

Comment:
This fixed it. what was happening was that t gets set and marked done at the top and this branch breaks  if done `int done = !task_state_any(q, VINE_TASK_RUNNING) && !task_state_any(q, VINE_TASK_READY) && !task_state_any(q, VINE_TASK_WAITING_RETRIEVAL);` so t would be returned as `NULL`